packages/CodeQuality/src/Rector/Foreach_/SimplifyForeachToCoalescingRector.php

```diff
@@ -149,15 +140,16 @@ private function processForeachNodeWithReturnInside(Foreach_ $foreachNode, Retur
 
         // is next node Return?
         if ($foreachNode->getAttribute(Attribute::NEXT_NODE) instanceof Return_) {
-            $this->returnToBeRemoved = $foreachNode->getAttribute(Attribute::NEXT_NODE);
+            $this->returnNode = $foreachNode->getAttribute(Attribute::NEXT_NODE);
+            $this->removeNode($this->returnNode);
         }
 
         $coalesceNode = new Coalesce(new ArrayDimFetch(
             $foreachNode->expr,
             $checkedNode
-        ), $this->returnToBeRemoved ? $this->returnToBeRemoved->expr : $checkedNode);
+        ), $this->returnNode ? $this->returnNode->expr : $checkedNode);
 
-        if ($this->returnToBeRemoved) {
+        if ($this->returnNode) {
             return new Return_($coalesceNode);
         }
 
```

Comment:
@carusogabriel Remove node in simpler way

Reply:
@TomasVotruba Thanks, I'll see if I can refactoring my implementation